js/doc.test.js

```diff
@@ -0,0 +1,99 @@
+    const all = Query.all(opts);
+    expect(all.offset()).toBe(10);
+    expect(all.limit()).toBe(10);
+
+    opts.direction = SortDirection.Desc;
+    opts.limit = 0;
+    opts.offset = 0;
+    const singleLatestPerKey = Query.singleLatestPerKey(opts);
+    expect(singleLatestPerKey.offset()).toBe(0);
+    expect(singleLatestPerKey.limit()).toBe(null);
+
+    opts.direction = SortDirection.Asc;
+    opts.offset = 100;
+    const author = Query.author(AuthorId.fromString("mqtlzayyv4pb4xvnqnw5wxb2meivzq5ze6jihpa7fv5lfwdoya4q"), opts);
+    expect(author.offset()).toBe(100);
+    expect(author.limit()).toBe(null);
+
+    opts.sortBy = SortBy.KeyAuthor;
+    opts.direction = SortDirection.Desc;
+    opts.offset = 0;
+    opts.limit = 100;
+    const keyExact = Query.keyExact(Buffer.from('key'), opts);
+    expect(keyExact.offset()).toBe(0);
+    expect(keyExact.limit()).toBe(100);
+
+    const keyPrefix = Query.keyPrefix(Buffer.from('prefix'), opts);
+    expect(keyPrefix.offset()).toBe(0);
+    expect(keyPrefix.limit()).toBe(100);
+});
+
+test('document entry basics', async () => {
+    const dir = path.join(require('os').tmpdir(), 'document_entry_basics');
+    const node = await IrohNode.withPath(dir);
+    const author = await node.authorCreate();
+    const doc = await node.docCreate();
+    const val = Buffer.from('hello world!');
+    const key = Buffer.from('foo');
+    const hash = await doc.setBytes(author, key, val);
+    const query = Query.authorKeyExact(author, key);
+    const entry = await doc.getOne(query);
+    expect(hash.equal(entry.contentHash())).toBe(true);
+    expect(val.length).toBe(entry.contentLen());
+    const gotVal = await entry.contentBytes(doc);
+    expect(gotVal).toEqual(val);
+});
+
+test('document import export', async () => {
+    const tmp = require('os').tmpdir();
+    const inRoot = require('path').join(tmp, "in");
+    const outRoot = require('path').join(tmp, "out");
+    require('fs').mkdirSync(inRoot, { recursive: true });
+    require('fs').mkdirSync(outRoot, { recursive: true });
+    const path = require('path').join(inRoot, "test");
+    const size = 100;
+    const bytes = Buffer.alloc(size).map(() => Math.floor(Math.random() * 256));
+    require('fs').writeFileSync(path, bytes);
+    const irohDir = require('os').tmpdir();
+    const node = await IrohNode.withPath(irohDir);
+    const doc = await node.docCreate();
+    const author = await node.authorCreate();
+    const key = pathToKey(path, null, inRoot);
+    let importProgress = await doc.importFile(author, key, path, true, null);
+    const iter = importProgress[Symbol.iterator]();
+    iter.return();
+    const query = Query.authorKeyExact(author, key);
+    const entry = await doc.getOne(query);
+    const outPath = keyToPath(key, null, outRoot);
+    let exportProgress = await doc.exportFile(entry, outPath);
+    const expIter = exportProgress[Symbol.iterator]();
```

Comment:
Oo

Reply:
pretty gross, huh? it's how you get the underlying iterator, so you can call `return`, which iterates through the whole generator & cleans it up. You can also do `for (const val of exportProgress)`, & just iterate through all the events, but I wanted to make sure `return` worked.